Task_Prototype/index.js

```diff
@@ -1,22 +1,66 @@
-//Write code for the 1st task below this
-
-
-//Write your code above this line
-//Write code for the 2nd task below this
-
-
-//Write your code above this line
-//Write code for the 3rd task below this
-
-
-//Write your code above this line
-//Write code for the 4th task below this
-
-
-//Write your code above this line
-//Write code for the 5th task below this
-
-
-//Write your code above this line
-
-
+//Write code for the 1st task below this
+function Person(name , age){
+    this.name = name;
+    this.age = age
+    this.greet = function(){
+         console.log(`Hello, ${this.name} you are ${this.age} years old.`);
+    }
+}
+
+//Write your code above this line
+//Write code for the 2nd task below this
+function Student(name, age, studentId) {
+  Person.call(this, name, age); 
+  this.studentId = studentId;
+}
+
+Student.prototype = Object.create(Person.prototype);
+
+Student.prototype.study = function() {
+  console.log(`${this.name} with student ID ${this.studentId} is studying.`);
+};
+
+//Write your code above this line
+//Write code for the 3rd task below this
+Person.prototype.introduce = function() {
+  console.log(`Hi, my name is ${this.name}.`);
+};
+
+// Demonstrate introduce method
+const person1 = new Person('John', 25);
+person1.introduce(); 
+
+const student1 = new Student('Alice', 20, '123456');
+student1.introduce(); 
+
+//Write your code above this line
+
+//Write code for the 4th task below this
+Student.prototype.greet = function() {
+  console.log(`Hey, I'm ${this.name} and I'm ${this.age} years old.`);

```

Comment:
Consider adding comments for better clarity on the advantages of using classes over the prototype-based approach.